src/state/page-models/actions.js

```diff
@@ -81,3 +84,67 @@ export const loadSelectedPageModel = pageCode => (dispatch, getState) => {
         return null;
       }));
 };
+
+export const fetchPageModel = pageModelCode => dispatch => (
+  new Promise((resolve, reject) => {
+    getPageModel(pageModelCode).then((response) => {
```

Comment:
please use fetchPageModel instead of  getPageModel also in loadSelectedPageModel